common/prettify.pm

```diff
@@ -618,47 +620,46 @@ sub Header ()
 
 sub Footer ()
 {
-    my $self = shift;
-    my $out = $self->{FH};
-
-    my $indent = '  ';
-    print $out $indent, "<testsuite name=\"Autobuild_Tests\" ";
-    if (defined $self->{TIMESTAMP} and length $self->{TIMESTAMP})
-    {
-        print $out "timestamp=\"$self->{TIMESTAMP}\" ";
-    }
-
-    my $numtests = @{$self->{TESTS}};
-
-    if ($numtests == 0)
-    {
-        # Need to insert a dummy testcase to jenkins doesn't think there is
-        # a failure.
-        print $out 'tests="1">', "\n", $indent x 2,
-          '<testcase name="dummy_test"/>', "\n";
-    }
-    else
-    {
-        print $out "tests=\"$numtests\" failures=\"$self->{FAILED}\">\n";
-    }
-
-    foreach my $test (@{$self->{TESTS}})
-    {
-        my $error = $test->{ERROR} || '';
-
-        print $out $indent x 2,
-          "<testcase name=\"$test->{NAME}\" status=\"$test->{RESULT}\" ",
-          "time=\"$test->{TIME}\"", ($error eq "" ? "/>\n" : '>');
-
-        if ($error ne "")
-        {
-            print $out "<failure>\n", $indent x 3,
-              "<![CDATA[$error]]></failure><system-out>\n", $indent x 3,
-              "<![CDATA[$test->{OUT}]]></system-out></testcase>\n";
-        }
-    }
-
-    print $out $indent, "</testsuite>\n</testsuites>\n";
+  my $self = shift;
+  my $out = $self->{FH};
+  my $log_file = new URI::URL(main::GetVariable('log_root') . '/' . main::GetVariable('log_file') . "_Full.html");
+
+  my $indent = '  ';
+  print $out $indent, "<testsuite name=\"Autobuild_Tests\" ";
+  if (defined $self->{TIMESTAMP} and length $self->{TIMESTAMP})
+  {
+    print $out "timestamp=\"$self->{TIMESTAMP}\" ";
+  }
+
+  my $numtests = @{$self->{TESTS}};
+  if ($numtests > 0) {
+    print $out "tests=\"$numtests\" failures=\"$self->{FAILED}\" hostname=\"$Prettify::Config_HTML::host\">\n";
+  }
+  else { # Insert a dummy testcase so jenkins doesn't think there is a failure.
+    print $out 'tests="1">', "\n", $indent x 2, '<testcase name="dummy_test"/>', "\n";
+  }
+
+  print $out $indent x 2, "<properties>\n";
+  print $out $indent x 3, "<property name=\"commits\" value=\"";
```

Comment:
For the sake of parsing these out later, I'd prefer the commits to have unique property names where the value was just the SHA and not lumped together in an array.

Reply:
replaced with hash